sources/engine/Stride.Assets.Models/ImportModelCommand.Animation.cs

```diff
@@ -64,7 +64,19 @@ private unsafe object ExportAnimation(ICommandContext commandContext, ContentMan
                     // No skeleton, map root node only
                     // TODO: For now, it seems to be located on node 1 in FBX files. Need to check if always the case, and what happens with Assimp
                     var rootNode0 = modelSkeleton.Nodes.Length >= 1 ? modelSkeleton.Nodes[0].Name : null;
-                    var rootNode1 = modelSkeleton.Nodes.Length >= 2 ? modelSkeleton.Nodes[1].Name : null;
+
+                    int counterLenCheck = 1;
+                    string rootNode1 = null;
+lable1CounterCheck:;
+                    ++counterLenCheck;
+                    if(modelSkeleton.Nodes.Length <counterLenCheck) { goto labelbreak; }
+                    rootNode1 = modelSkeleton.Nodes.Length >= counterLenCheck ? modelSkeleton.Nodes[counterLenCheck-1].Name : null;
+                    if(rootNode1 == null || rootNode1.ToLower().Contains("$assimp")) 
+                    { 
+                        goto lable1CounterCheck; 
+                    }
+labelbreak:;
```

Comment:
This could be converted into a do while loop afaict:
```cs
int counterLenCheck = 1;
string rootNode1 = null;
do
{
	++counterLenCheck;
	if(modelSkeleton.Nodes.Length <counterLenCheck)
		break;
	rootNode1 = modelSkeleton.Nodes.Length >= counterLenCheck ? modelSkeleton.Nodes[counterLenCheck-1].Name : null;
} while(rootNode1 == null || rootNode1.ToLower().Contains("$assimp"));
```

Reply:
fixed!